diff --git a/tests/socket.tst b/tests/socket.tst
index 9f8940033..09b0a6339 100644
--- a/tests/socket.tst
+++ b/tests/socket.tst
@@ -670,14 +670,14 @@ T
     (ext:run-program run
                      :arguments (append args
                                   (list "-q" "-q" "-x"
-                                        (format nil "(close (prog1 (socket:socket-connect ~D) (sleep 0.01s0)))" (socket:socket-server-port se))))
+                                        (format nil "(close (prog1 (socket:socket-connect ~D) (sleep 0.1s0)))" (socket:socket-server-port se))))
                      :wait nil :input nil :output nil)
     (unwind-protect
         (with-open-stream (so (socket:socket-accept se))
           (list
             (socket:socket-status so)
             (write-line "foo" so)
-            (sleep 0.02s0)
+            (sleep 0.2s0)
             (handler-case (socket:socket-status so)
               (os-error (c)
                 (princ 'socket-status-2) (princ-error c)
